### PR TITLE
Fix Gitlab plugin getting automatic repoid from reponame

### PR DESCRIPTION
**Pull request by SRombauts**
- listing all projects of a user was not working is the user was not explicitly added to every group or repository
 - accessing a project by its name is availlable to every user with at least a read access,
   so this method is valid for every Public or Internal project
 - works with the latest Gitlab I installed in our company, with the instruction based on readme

### CODE REVIEW

File: SourceGitlab/SourceGitlab.php [hunk 1]
```diff
@@ -98,9 +98,9 @@ public function url_diff( $p_repo, $p_changeset, $p_file ) {
 	}
 
 public function update_repo_form( $p_repo ) {
-		$t_hub_root = null;
+		$t_hub_root = "http://ach-gitlab";
 		$t_hub_repoid = null;
-		$t_hub_app_secret = null;
+		$t_hub_app_secret = "xuXcB5xgZgognLN96GmN";
```
Comment (dregad) [inline SourceGitlab/SourceGitlab.php]: Erm... is that intentional ?

**Comment by SRombauts** [comment]
Sorry again, it seems I left to much test code everywhere!

I made many temporary modification interleaved with a few fixes, then I tried to extract each change in a comprehensible pull request.
So here is the core of the real change.

This is basically a change of API from GET /projects" to "GET /projects/NAMESPACE/PROJECT", so getting details from only one project instead of an array of all projects that the user is part of (which is not working for a public/internal project the user is not into).
See [API ref](http://doc.gitlab.com/ce/api/projects.html)

File: SourceGitlab/SourceGitlab.php [hunk 2]
```diff
@@ -147,38 +147,35 @@ public function update_repo( $p_repo ) {
...
-		if( empty( $f_hub_repoid ) ) {
+		# Getting the repoid from reponame
+		if( !is_numeric( $f_hub_repoid ) && !empty( $f_hub_reponame ) ) {
 			$t_hub_reponame_enc = urlencode( $f_hub_reponame );
-			$t_uri = $this->api_uri( $p_repo, "projects" );
+			$t_uri = $this->api_uri( $p_repo, "projects/$t_hub_reponame_enc" );
 			$t_member = null;
 			$t_json = json_url( $t_uri, $t_member );
 
-			$f_hub_repoid = 'RepoName is invalid';
```
Comment (umang-orbit) [inline SourceGitlab/SourceGitlab.php]: test

Comment (umang-orbit) [inline SourceGitlab/SourceGitlab.php]: Hi I am getting this error on adding gitlab repo. No help available online.
Please help. I will delete this comment after one sees this. I am not giving anything in API key

http://192.168.0.205/umang/sample.git
array (size=5)
  'hub_root' => string 'http://192.168.0.205' (length=20)
  'hub_repoid' => string 'RepoName is invalid' (length=19)
  'hub_reponame' => string 'umang/sample' (length=12)
  'hub_app_secret' => string '' (length=0)
  'master_branch' => string 'master' (length=6)

**Pull request closed without merging**